### PR TITLE
Passwordless

### PR DESCRIPTION
**Pull request by lambdaR**
Summary of changes to #250:

- SendMagicLinkRequest has two new fields namely, address and endpoint
- SendMagicLink will do the necessary checks and send a MagicLink to the specified email, the MgicLink format will be `https://address/endpoint/token`
- VerifyTokenResponse has two fields now, isValid and message
- When isValid is true, it means that a session object has been created and published.

@asim, It would be convenient if there was a minimal M3O back-end replica that we can run locally so we can test our contributions against it.

### CODE REVIEW

**Comment by asim** [comment]
What's the reason for subscribing to the broken when sending the magic link? Can we just have the endpoint send the link and not worry about anything else?

**Comment by lambdaR** [comment]
See the Gif here at https://magic.link/ this is what I want to achieve, a similar UX.
front-end/Mobile apps developers can use these two endpoints to mimic that same UX sort of.

Can we just have the endpoint send the link and not worry about anything else? Yes we could, that's even simpler. Your call.

**Comment by asim** [comment]
You'll need to add examples for each endpoint to examples.json

File: user/proto/user.proto [hunk 1]
```diff
@@ -237,3 +239,35 @@ message ListRequest {
...
+	Session session = 1;
+}
+
+// Verify the token sent via a magic link
+message VerifyTokenRequest {
+	string token = 1;
+}
+
+message VerifyTokenResponse {
+	bool isvalid = 1;
```
Comment (asim) [inline user/proto/user.proto:271]: Change to is_valid

Comment (lambdaR) [reply user/proto/user.proto:271]: I followed what already been used here for naming (camel case).
it should be snake case instead.

**Comment by asim** [comment]
What's your email on m3o.com? I'll credit you $5

**Pull request merged**